knife/lib/chef/utils/licensing_handler.rb

```diff
@@ -30,9 +30,16 @@ def install_sh_url
 
       class << self
         def validate!
-          license_keys = ChefLicensing::LicenseKeyFetcher.fetch
-
-          return new(nil, nil) if license_keys.blank?
+          license_keys = begin
+                           ChefLicensing::LicenseKeyFetcher.fetch
+                         # If the env is airgapped or the local licensing service is unreachable,
+                         # the licensing gem will raise ChefLicensing::RestfulClientConnectionError.
+                         # In such cases, we are assuming the license is not available.
+                         rescue ChefLicensing::RestfulClientConnectionError
+                           []
+                         end
+
+          return new(nil, nil) if license_keys.nil? || license_keys.empty?
```

Comment:
This can be simplified to use safe navigation that checks for nil before trying method empty. If nil it'll return nill which is then false in the if statement.
```suggestion
          return new(nil, nil) if license_keys&.empty?
```